object_templates/provider_type.erb

```diff
@@ -3,8 +3,16 @@ require 'puppet/resource_api'
 Puppet::ResourceApi.register_type(
   name: '<%= name %>',
   docs: <<-EOS,
-      This type provides Puppet with the capabilities to manage ...
-    EOS
+@summary Fancy new type
+@example Fancy new example
```

Comment:
```
@example 
<%= name %> { 'foo':
ensure: present,
name: 'foo'
}
```

Reply:
We could foo it up or use a real example like apt_key. I'm not particularly attached to either.

Reply:
As a help to the developer, this should be as close as possible to the final result as we can get it, so `@example <%= name %> { 'foo': ...` is much better than something with `apt_key`, as it will be technically correct at the time of rendering, and not confuse people, e.g. in the generated documentation.